resources/ansible/roles/downloaders/files/download.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+CONTACT_PEER="${1:-}"
+
+CONTACT_PEER_ARG=""
+if [[ -n "$CONTACT_PEER" ]]; then
+  CONTACT_PEER_ARG="--peer $CONTACT_PEER"
+fi
+
+if ! command -v safe &> /dev/null; then
+  echo "Error: 'safe' not found in PATH."
+  exit 1
+fi
+
+LOG_FILE="./uploaded_files.log"
+DOWNLOAD_DIR="./downloaded_files"
+SLEEP_INTERVAL=5
+
+mkdir -p "$DOWNLOAD_DIR"
+
+download_file() {
+  local file_ref=$1
+  # Multiple downloaders can be running on the same machine, so one or more
+  # could select the same file address at the same time. We therefore use a GUID
+  # to ensure a unique output file.
+  local output_path="$(uuidgen).dat"
+  (
+    cd "$DOWNLOAD_DIR"
+    echo "Downloading file: $file_ref"
+    safe $CONTACT_PEER_ARG files download "$output_path" "$file_ref"
```

Comment:
Could we clean up the file after the download. We might quickly run out of space